cli/dcoscli/node/main.py

```diff
@@ -65,12 +75,357 @@ def _cmds():
+    """
+    Manage snapshots
+
+    :param list_snapshots: a list of available snapshots
+    :type  list_snapshots: bool
+    :param status: show snapshot job status
+    :type  status: bool
+    :param cancel: cancel snapshot job
+    :type  cancel: bool
+    :return: process return code
+    :rtype: int
+    """
+
+    _check_3dt_version()
+    if list_snapshots:
+        if json:
+            emitter.publish(_get_snapshots_json())
+            return 0
+
+        available_snapshots = _get_snapshots_list()
+        if not available_snapshots:
+            emitter.publish("No snapshots")
+            return 0
+        emitter.publish("Available snapshots:")
+        for available_snapshot in sorted(available_snapshots,
+                                         key=lambda t: t[0]):
+            emitter.publish('{} {}'.format(available_snapshot[0],
+                                           sizeof_fmt(available_snapshot[1])))
+        return 0
+    elif status:
+        url = urllib.parse.urljoin(SNAPSHOT_BASE_URL, 'status/all')
+        snapshot_response = _do_snapshot_request(url, 'GET')
+        if json:
+            emitter.publish(snapshot_response)
+            return 0
+
+        for host, props in sorted(snapshot_response.items()):
+            emitter.publish(host)
+            for key, value in sorted(props.items()):
+                emitter.publish('  {}: {}'.format(key, value))
+            emitter.publish('\n')
+        return 0
+    elif cancel:
+        url = urllib.parse.urljoin(SNAPSHOT_BASE_URL, 'cancel')
+        snapshot_response = _do_snapshot_request(url, 'POST')
+        if json:
+            emitter.publish(snapshot_response)
+            return 0
+
+        if 'status' not in snapshot_response:
+            raise DCOSException(
+                'Request to cancel a snapshot job {} returned '
+                'an unexpected response {}'.format(url, snapshot_response))
+
+        emitter.publish(snapshot_response['status'])
+        return 0
+    else:
+        raise DCOSException(
+            'Must specify one of  list_snapshots, status, cancel')
+
+
+@snapshot_error
+def _do_request(url, method, timeout=3, stream=False, **kwargs):
+    """
+    make HTTP request
+
+    :param url: url
+    :type url: string
+    :param method: HTTP method, GET or POST
+    :type  method: string
+    :param timeout: HTTP request timeout, default 3 seconds
+    :type  timeout: integer
+    :param stream: stream parameter for requests lib
+    :type  stream: bool
+    :return: http response
+    :rtype: requests.Response
+    """
+
+    def _is_success(status_code):
+        # consider 400 and 503 to be successful status codes.
+        # API will return the error message.
+        if status_code in [200, 400, 503]:
+            return True
+        return False
+
+    # if timeout is set to None, pass it to requests,
+    # otherwise try to use a timeout from a config, or default to 3 seconds.
+    if timeout is not None:
```

Comment:
timeout will never be None because you set it to 3, I think you want to default it to None, and then have timeout = timeout or config.get_config_val('core.timeout') or 3